libseccomp/src/notify.rs

```diff
@@ -5,107 +5,299 @@
 
 use crate::error::ErrorKind::*;
 use crate::error::{Result, SeccompError};
+use crate::{ScmpArch, ScmpFilterContext};
 use libseccomp_sys::*;
-use std::ptr::NonNull;
+use std::io::Error;
 
-pub struct ScmpNotification {
-    req: NonNull<seccomp_notif>,
-    resp: NonNull<seccomp_notif_resp>,
+#[doc(hidden)]
+fn get_errno() -> i32 {
+    if let Some(errno) = Error::last_os_error().raw_os_error() {
+        errno
+    } else {
+        0
+    }
 }
 
-impl ScmpNotification {
-    pub fn new() -> Result<Self> {
-        let mut req_ptr: *mut seccomp_notif = std::ptr::null_mut();
-        let mut resp_ptr: *mut seccomp_notif_resp = std::ptr::null_mut();
+/// ScmpFd represents a file-descriptor used for seccomp userspace notifications.
+type ScmpFd = i32;
```

Comment:
```diff rust
+ use std::os::unix::io::RawFd;
```

```suggestion
type ScmpFd = RawFd;
```
- Just as an idea, it doesn't make any difference.
- Is this type definition private by intention? IMHO it should be `pub`lic as it is returned by functions.

Reply:
Thank you. I forgot to define `ScmpFd` as  `pub`. I fixed it.